lib/ex_unit_properties.ex

```diff
@@ -222,7 +222,12 @@ defmodule ExUnitProperties do
     * value generation clauses affect shrinking similarly to `bind/2`
 
   """
-  # TODO: support do: syntax
+  defmacro gen({:all, _meta, clauses_with_body}) do
+    [[do: body] | reversed_clauses] = Enum.reverse(clauses_with_body)
```

Comment:
`{clauses, [[do: body]]} = Enum.drop(clauses_with_body, -1)` would do it too.

Reply:
Good catch!

Reply:
I don't think this would work, since `Enum.drop([1, 2, 3], -1) # => [1, 2]` and you would lose the body data. https://hexdocs.pm/elixir/Enum.html#drop/2 I was thinking of doing it the same way, by using drop, but saw that it didn't return the dropped element.

Reply:
> (although you meant split :P)

:P